server.go

```diff
@@ -527,14 +527,33 @@ func (r *oauthProxy) createHTTPListener(config listenerConfig) (net.Listener, er
 		}
 
 		tlsConfig := &tls.Config{
-			GetCertificate:           getCertificate,
+			GetCertificate: getCertificate,
+			// Causes servers to use Go's default ciphersuite preferences,
+			// which are tuned to avoid attacks. Does nothing on clients.
 			PreferServerCipherSuites: true,
-			NextProtos:               []string{"h2", "http/1.1"},
+			// Only use curves which have assembly implementations
+			// https://github.com/golang/go/tree/master/src/crypto/elliptic
+			CurvePreferences: []tls.CurveID{
+				tls.CurveP256,
+				tls.X25519,
+			},
+			NextProtos: []string{"http/1.1", "h2"},
+			// https://www.owasp.org/index.php/Transport_Layer_Protection_Cheat_Sheet#Rule_-_Only_Support_Strong_Protocols
+			MinVersion: tls.VersionTLS12,
+			// Use modern tls mode https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+			// See security linter code: https://github.com/securego/gosec/blob/master/rules/tls_config.go#L11
+			// These ciphersuites support Forward Secrecy: https://en.wikipedia.org/wiki/Forward_secrecy
+			CipherSuites: []uint16{
+				tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
```

Comment:
That is a breaking change for old browsers/clients. A hardcoded list of ciphers will be also a problem for new TLS 1.3 ciphers (hopefully available from Go 1.12). IMHO better approach is to allow TLS configuration on the user level: --cipher-suites, --tls-min-version, --tls-prefer-server-cipher-suites, .... In this case, a user will be able to define, what kind of TLS level will be supported (old browsers, only selected ciphers, ...). Default Golang TLS configuration can be used as default TLS configuration.

Reply:
@jangaraj added a commit along the lines of your proposal